python/translate_source_bone.py

```diff
@@ -0,0 +1,415 @@
+    # else:
+    #     last_leg_item = l_t_index
+    #     last_leg_name = last_leg_name_cmp_l
+    # # print(f"This is last leg item {old_last_leg_item}")
+
+    # base bone section
+    # base order: 上半身, 下半身, 腰 (b_1), グルーブ, センター, 全ての親
+    b1_name = "腰"
+    b2_name = "グルーブ"
+    b3_name = "センター"
+    b4_name = "全ての親"
+
+    # IK bone section
+    leg_left_ik_name = "左足ＩＫ"
+    leg_left_toe_ik_name = "左つま先ＩＫ"
+    leg_right_ik_name = "右足ＩＫ"
+    leg_right_toe_ik_name = "右つま先ＩＫ"
+
+    knee_limit_1 = [-3.1415927410125732, 0.0, 0.0]
+    knee_limit_2 = [-0.008726646192371845, 0.0, 0.0]
+
+    # for some reasons, this value will always be the same
+    # pelvis_pos = [-4.999999873689376e-06, 38.566917419433594, -0.533614993095398]
+
+    # adding IK and such
+    # leg_left_obj = retme.bones[last_leg_item + l_l]
+    # leg_left_knee_obj = retme.bones[last_leg_item + l_k]
+    leg_left_ankle_obj = retme.bones[l_a_index]
+    leg_left_toe_obj = retme.bones[l_t_index]
+    # leg_right_obj = retme.bones[last_leg_item + r_l]
+    # leg_right_knee_obj = retme.bones[last_leg_item + r_k]
+    leg_right_ankle_obj = retme.bones[r_a_index]
+    leg_right_toe_obj = retme.bones[r_t_index]
+
+    leg_left_ankle_pos = leg_left_ankle_obj.pos
+    leg_left_toe_pos = leg_left_toe_obj.pos
+    leg_right_ankle_pos = leg_right_ankle_obj.pos
+    leg_right_toe_pos = leg_right_toe_obj.pos
+
+    # toe /// places of some value wont match with the struct /// taken from hololive's korone model
+    # name, name, [-0.823277473449707, 0.2155265510082245, -1.8799238204956055], 112, 0, False,
+    # True, True, True, True,
+    # False, [0.0, -1.3884940147399902, 1.2653569569920364e-07] /// This is offset, False, False, None,
+    # None, False, None, False, None, None, False, None, True,
+    # 111, 160, 1.0, [[110, None, None]]
+
+    # leg
+    # 右足ＩＫ, en_name, [-0.8402935862541199, 1.16348397731781, 0.3492986857891083], 0, 0, False,
+    # True, True, True, True,
+    # False, [0.0, -2.53071505085245e-07, 1.3884940147399902], False, False, None,
+    # None, False, None, False, None, None, False, None, True,
+    # 110, 85, 1.9896754026412964, [[109, [-3.1415927410125732, 0.0, 0.0], [-0.008726646192371845, 0.0, 0.0]]
+    # /// These ik_links are in radians /// , [108, None, None]]
+    # if name == "ValveBiped.Bip01_R_Toe0":
+    #     retme.bones.insert(last_leg_item + 1, )
+
+    leg_left_ik_obj = pmxstruct.PmxBone(leg_left_ik_name, "", leg_left_ankle_pos, last_bone + 5, 0, False,
+                                        True, True, True, True, True,
+                                        False, [0.0, 0.0, 0.0], False, False, False,
+                                        False, False, None, None, None, None, None, None,
+                                        l_a_index, 40, 114.5916,
+                                        [[l_k_index, knee_limit_1, knee_limit_2],
+                                         [l_l_index, None, None]])
```

Comment:
This is why you thought PMX parser was bugged, repeat for other 3 IK bones
```suggestion
                                        [pmxstruct.PmxBoneIkLink(idx=l_k_index, limit_min=knee_limit_1, limit_max=knee_limit_2), 
                                         pmxstruct.PmxBoneIkLink(idx=l_l_index)])
```